### DISCUSSION TITLE
大佬救！！！文章的创建时间如何使用最后修改时间？

### DISCUSSION THREAD

**Discussion by sizdshi** [MaLuns/hexo-theme-async#91] [Q&A 疑难杂症] [answered]
我之前的md文件因为存放在git上，所以创建时间都是最新的。我按网上教程将文件最新修改时间改回后，发现主题里还是使用的是，文章的创建时间，而不是修改时间。

已经尝试改过updated_option: "mtime"参数，不起作用。
也已在scaffolds/post.md文件中添加updated: {{ date }}，也不起作用。

想请问这个问题怎么解决？以下是截图。

![4229813cc5588f5f5e128235c42983e](https://github.com/MaLuns/hexo-theme-async/assets/30177132/28c724d7-107c-42ca-bfc9-831a7151b13e)
![dd98e08422f33e1d0284e5ae449e920](https://github.com/MaLuns/hexo-theme-async/assets/30177132/a22c30be-caaa-4052-8c81-d2ae6b994342)

**Comment by MaLuns** [comment]
看了下代码，主题里默认用的创建时间🤣，后面优化下

**Reply by sizdshi** [reply]
!!!谢谢佬！！！我这有5，60个md文件，一个个改date真的太痛了！！🤕

**Reply by sizdshi** [reply]
主题真的很好！！！！！打call ！！！！！！

**Reply by MaLuns** [reply]
发了个小版本，可以自己配置日期选择用哪个，可以参考这里 https://hexo-theme-async.imalun.com/guide/config.html#%E6%97%A5%E6%9C%9F%E6%A0%BC%E5%BC%8F%E5%8C%96-date-format

**Reply by sizdshi** [reply]
大佬，我试了一下，好像没效果，是我哪里配的有问题吗？

[这是我的个人网址：](https://www.tctech.asia)
![66c18169a35a808d037b54c59d83e1f](https://github.com/MaLuns/hexo-theme-async/assets/30177132/ed989573-bf8b-40b5-8ae6-9249311f3480)

**Reply by sizdshi** [reply]
大佬，还有这个文章的标题是都需要自己取是吗？不是自动根据文件名获得的那种。

**Reply by MaLuns** [reply, marked as answer]
你应该配置 post_card 和 post_info 里的

**Reply by sizdshi** [reply]
是我蠢了😭！！大佬你太厉害啦！！！！打call！！！！！！！！

**Reply by sizdshi** [reply]
大佬，那个文章名是要自己在md文件中写title吗？一写的话，就改了文件的最后修改时间，有根据文件名自动设置的配置方法吗？文档里没找到😭😭

**Reply by sizdshi** [reply]
归档里文章排序，时间显示，和首页文章排序好像还是按照文件的创建时间排序的。

突然感觉好麻烦大佬！像是在提需求😱😱😱

还是我自己手动先改改！不麻烦大佬！dalao你想改就改，不想修就不修！能用到这么好的主题，已经很满足啦！！！

再次谢谢大佬的精美主题！！！

![44ece5c4b8801824949fc838ba9dcf1](https://github.com/MaLuns/hexo-theme-async/assets/30177132/90aa623a-7bc1-4879-8d2b-911647abb8f2)

**Reply by MaLuns** [reply]
这个需要覆盖 hexo 内置生成这个归档列表的辅助函数了，标题那个按照hexo 文档也是默认取 md 里 title 的，你想改的话可以通过 hexo 提供的扩展api，可以在生成前替换变量

**Comment by MaLuns** [comment]
排序那个误导你了, hexo 本身是有可以配置排序的 , 在 hexo 配置文件里 index_generator 和 archive_generator 下配置 order_by 字段

**Reply by sizdshi** [reply]
收到，我去研究了看看！谢谢大佬！！！

**Reply by sizdshi** [reply]
对了，大佬你的官方文档网页挂了，在这个点。
